pkg/cache/tas_flavor_snapshot.go

```diff
@@ -803,22 +910,29 @@ func (s *TASFlavorSnapshot) sortedDomains(domains []*domain, unconstrained bool)
 	isLeastFreeCapacity := useLeastFreeCapacityAlgorithm(unconstrained)
 	result := slices.Clone(domains)
 	slices.SortFunc(result, func(a, b *domain) int {
-		if a.state == b.state {
-			return slices.Compare(a.levelValues, b.levelValues)
+		if a.sliceState != b.sliceState {
+			if isLeastFreeCapacity {
+				// Start from the domain with the least amount of free resources.
+				// Ascending order.
+				return cmp.Compare(a.sliceState, b.sliceState)
+			}
+			return cmp.Compare(b.sliceState, a.sliceState)
 		}
-		if isLeastFreeCapacity {
-			// Start from the domain with the least amount of free resources.
-			// Ascending order.
+
+		if a.state != b.state {
 			return cmp.Compare(a.state, b.state)
 		}
-		// Descending order.
-		return cmp.Compare(b.state, a.state)
+
+		return slices.Compare(a.levelValues, b.levelValues)
 	})
+
```

Comment:
nit, I would suggest avoiding drive-by formatting changes. Sometimes it adds extra work on cherry-picks.